AK/ByteBuffer.h

```diff
@@ -134,13 +134,14 @@ class ByteBuffer {
     [[nodiscard]] void* end_pointer() { return data() + m_size; }
     [[nodiscard]] void const* end_pointer() const { return data() + m_size; }
 
-    // FIXME: Make this function handle failures too.
-    [[nodiscard]] ByteBuffer slice(size_t offset, size_t size) const
+    [[nodiscard]] ErrorOr<ByteBuffer> slice(size_t offset, size_t size) const
     {
         // I cannot hand you a slice I don't have
-        VERIFY(offset + size <= this->size());
+        if (offset + size <= this->size()) {
+            return Error::from_string_literal("ByteBuffer: Out of Range");
+        };
```

Comment:
This doesn't seem desirable - passing an invalid size/offset combination is still a programming error and should crash the program.